addon/model.js

```diff
@@ -100,8 +107,7 @@ function resolveValue(key, value, modelName, store, schema, model, valueInArray
     return resolvePlainArray(key, value, modelName, store, schema, model);
   }
 
-  let reference = schema.computeAttributeReference(key, value, modelName, schemaInterface);
-
+  let reference = _computeAttributeReference(key, value, modelName, schemaInterface, schema);
```

Comment:
From correctness perspective, we need to track the unknown keys for every function in `schema-manager`, which we pass the `schemaInterface` to. Right now, we really only need it for `computeAttributeReference`, because of our specific implementation of the Schema.